funtofem/optimization/utils/plot_manager.py

```diff
@@ -0,0 +1,188 @@
+                    hist_dict[plot_name] = value
+                hist_dict_list += [hist_dict]
+        print(f"sample hist dict = {hist_dict_list[0]}")
+        return cls(hist_dict_list=hist_dict_list)
+
+    def __init__(self, hist_dict_list):
+        """
+        history array
+        """
+        self.hist_dict_list = hist_dict_list
+        self.functions = []  # funtofem functions objects
+        self.constr_dicts = []
+
+    def include(self, func):
+        self.functions += [func]
+
+    def valid_function(self, func):
+        return func.name in self.hist_dict_list[0]
+
+    def get_hist_array(self, func):
+        array = np.array([hist_dict[func.name] for hist_dict in self.hist_dict_list])
+        return array * func.scale
+
+    def get_constr_array(self, func):
+        final_value = self.hist_dict_list[-1][func.name]
+        constr_value = None
+        if func.lower is not None and func.upper is not None:
+            print(f"mode 1")
+            lower_dist = abs(final_value - func.lower)
+            upper_dist = abs(final_value - func.upper)
+            if lower_dist < upper_dist:
+                constr_value = func.lower * func.scale
+            else:
+                constr_value = func.upper * func.scale
+        elif func.lower is not None:
+            constr_value = func.lower * func.scale
+        elif func.upper is not None:
+            constr_value = func.upper * func.scale
+        assert constr_value is not None
+        return np.array([constr_value for _ in self.iterations])
+
+    def add_constraint(self, name: str, value: float = 0.0):
+        self.constr_dicts += [
+            {
+                "name": name,
+                "value": value,
+            }
+        ]
+
+    def get_r_string(self, omag):
+        assert isinstance(omag, int)
+        if omag == 0:
+            return ""
+        elif omag == 1:
+            return r"*$10^1$"
+        elif omag == 2:
+            return r"*$10^2$"
+        elif omag == 3:
+            return r"*$10^3$"
+        elif omag == 4:
+            return r"*$10^4$"
+        elif omag == 5:
+            return r"*$10^5$"
+        elif omag == 6:
+            return r"*$10^6$"
+        elif omag == -1:
+            return r"*$10^{-1}$"
+        elif omag == -2:
+            return r"*$10^{-2}$"
+        elif omag == -3:
+            return r"*$10^{-3}$"
+        elif omag == -4:
+            return r"*$10^{-4}$"
+        elif omag == -5:
+            return r"*$10^{-5}$"
+        elif omag == -6:
+            return r"*$10^{-6}$"
+
+    @property
+    def iterations(self) -> list:
+        return [_ for _ in range(len(self.hist_dict_list))]
+
+    def __call__(
+        self,
+        plot_name=None,
+        show_scales=False,
+        legend_frac: float = 0.8,
+        yaxis_name="func values",
+        color_offset: int = 0,
+    ):
+        if plot_name is None:
+            plot_name = "f2f-history"
+        plt.style.use(niceplots.get_style())
```

Comment:
If you want to avoid forcing the niceplots style on plots made outside of this function, you could use the context manager version of this:

`with plt.style.context(niceplots.get_style()):`

Reply:
Thanks for the tip, I implemented it.